plugins/builtin/strategy/threshold/plugin/plugin.go

```diff
@@ -0,0 +1,296 @@
+		if s != "" {
+			nonEmpty++
+		}
+	}
+	if nonEmpty == 0 {
+		return nil, fmt.Errorf("missing required field, must have either %q, %q or %q", runConfigKeyDelta, runConfigKeyPercentage, runConfigKeyValue)
+	}
+	if nonEmpty != 1 {
+		return nil, fmt.Errorf("only one of %q, %q or %q must be provided", runConfigKeyDelta, runConfigKeyPercentage, runConfigKeyValue)
+	}
+
+	if deltaStr != "" {
+		// Delta must be an interger.
+		d, err := strconv.ParseInt(deltaStr, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%q value %v is not an interger", runConfigKeyDelta, deltaStr)
+		}
+
+		c.actionType = runConfigKeyDelta
+		c.actionValue = float64(d)
+	}
+
+	if percentageStr != "" {
+		// Percentage must be a float.
+		p, err := strconv.ParseFloat(percentageStr, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%q value %v is not a number", runConfigKeyPercentage, percentageStr)
+		}
+
+		c.actionType = runConfigKeyPercentage
+		c.actionValue = p
+	}
+
+	if valueStr != "" {
+		// Value must be a positive interger.
+		v, err := strconv.ParseInt(valueStr, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("%q value %v is not an interger", runConfigKeyValue, valueStr)
+		}
+		if v < 0 {
+			return nil, fmt.Errorf("%q value %v is negative", runConfigKeyValue, valueStr)
+		}
+
+		c.actionType = runConfigKeyValue
+		c.actionValue = float64(v)
+	}
+
+	return c, nil
+}
+
+// parseBound parses and validates the value for a bound.
+func parseBound(bound string, input string) (float64, error) {
+	var defaultValue float64
+
+	switch bound {
+	case runConfigKeyLowerBound:
+		defaultValue = -math.MaxFloat64
+	case runConfigKeyUpperBound:
+		defaultValue = math.MaxFloat64
+	}
+
+	if input == "" {
+		return defaultValue, nil
+	}
+
+	value, err := strconv.ParseFloat(input, 64)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value for %q: %v (%T)", bound, input, input)
+	}
+
+	return value, nil
+}
+
+// withinBounds returns true if the metric result is considered within bounds.
+func withinBounds(logger hclog.Logger, metrics sdk.TimestampedMetrics, config *thresholdPluginRunConfig) bool {
+	logger.Trace("checking how many data points are within bounds")
+
+	withinBoundsCounter := 0
+	for _, metric := range metrics {
+		if metric.Value >= config.lowerBound && metric.Value < config.upperBound {
+			withinBoundsCounter++
+		}
+	}
+
+	logger.Trace(fmt.Sprintf("found %d data points within bounds", withinBoundsCounter))
+	return withinBoundsCounter >= config.withinboundsTrigger
+}
+
+// runDelta returns the next count for a delta check.
+func runDelta(count int64, d float64) int64 {
+	return count + int64(d)
+}
+
+// runPercentage returns the next count for a percentage check.
+func runPercentage(count int64, pct float64) int64 {
+	newCount := float64(count) * (1 + pct/100)
+	return int64(math.Ceil(newCount))
+}
+
+// runValue returns the next count for a value check.
+func runValue(count int64, v float64) int64 {
+	return int64(v)
+}
```

Comment:
It doesn't seem that the `count` argument is used; is this expected? If so can we remove it from the function signature?

Reply:
At some point there was a common function type for these three functions, but I thought it would couple them too much. I missed removing it here 👍